pkg/metadata/manager.go

```diff
@@ -28,29 +34,54 @@ type Manager interface {
 	// AllocateArtifactFilePath allocates the file path of the artifact.
 	AllocateArtifactFilePath(src *artifacts.Source, installBinary bool) (string, error)
 
+	// AllocateClusterScopeDirs allocates the directories of one cluster.
+	AllocateClusterScopeDirs() error
+
+	// AllocateClusterConfigPath allocates the config path of one cluster.
+	AllocateClusterConfigPath(cfg *config.Config) error
+
 	// SetHomeDir sets the home directory of the metadata manager.
 	SetHomeDir(dir string) error
 
 	// GetWorkingDir returns the working directory of the metadata manager.
 	// It should be ${HomeDir}/${BaseDir}.
 	GetWorkingDir() string
 
+	// GetClusterScopeDir returns the cluster scope directory of current cluster.
+	GetClusterScopeDir() *ClusterScopeDir
+
 	// Clean cleans up all the metadata. It will remove the working directory.
 	Clean() error
 }
 
 const (
-	// BaseDir is the working directory of gtctl and all the metadata will be stored in ${HomeDir}/${BaseDir}.
+	// BaseDir is the working directory of gtctl and
+	// all the metadata will be stored in ${HomeDir}/${BaseDir}.
 	BaseDir = ".gtctl"
+
+	// Default cluster scope directories.
+	clusterLogsDir = "logs"
+	clusterDataDir = "data"
+	clusterPidsDir = "pids"
 )
 
+type ClusterScopeDir struct {
+	BaseDir    string
+	LogsDir    string
+	DataDir    string
+	PidsDir    string
+	ConfigPath string
+}
+
 type manager struct {
 	workingDir string
+
+	clusterDir *ClusterScopeDir
 }
 
 var _ Manager = &manager{}
 
-func New(homeDir string) (Manager, error) {
+func New(homeDir string, clusterName string) (Manager, error) {
```

Comment:
The metadata manager should be decoupled from the cluster concept. I think we should add an interface like `CreateClusterMetadata(clusterName)` instead of adding `clusterDir`.

Reply:
i will refactor it.

now the `Allocate*` interface is in charge of assign the dirs or path (they dont actually create it), and add a new `Create` interface to create them